monte-carlo-method/README.md

```diff
@@ -0,0 +1,58 @@
+Table of Contents
+=================
+
+* [Monte Carlo method implementation examples](#monte-carlo-method-implementation-examples)
+* [Usage](#usage)
+* [PI Implementation](#pi-implementation)
+   * [MultiVertex implementation (vertex_ipu_pi)](#multivertex-implementation-vertex_ipu_pi)
+   * [Poplibs implementatio (poplibs_ipu_pi)](#poplibs-implementatio-poplibs_ipu_pi)
+   * [Iterative implementation (iterative_ipu_pi)](#iterative-implementation-iterative_ipu_pi)
+* [Integrals implementation](#integrals-implementation)
+   * [MultiVertex implementation (vertex_ipu_*)](#multivertex-implementation-vertex_ipu_)
+
+# Monte Carlo method implementation examples
+
+Example programs to estimate PI and integral value using Monte Carlo method
+for IPU architecture.
+
+# Usage
+
+All binaries build in these examples take up to 4 parameters:
+```
+--iterations <number> (default 30.000.000) defines number of <chunk_size> check loops
+--chunk_size <number> (default 10.000.000) defines size of the buffer for samples to be checked
+--num_ipus  <number>  (default 1)          defines how many IPUs will be used for computation
+--precision <number>  (default 10)         defines number of digits of the result number to be printed
+```
+
+# PI Implementation
+
+Every implementation generates random point and check it against x^2 + y^2 <= 1 and increments count `--iterations` times.
```

Comment:
Every implementation generates `--iterations` number of points with random coordinates x and y. Every point is checked against circle equation x^2 + y ^2 <=1 and when it is fulfilled counter is incremented.